lib/ansible/plugins/connection/ssh.py

```diff
@@ -818,12 +818,17 @@ def _bare_run(self, cmd, in_data, sudoable=True, checkrc=True):
                 p = None
 
         if not p:
-            if PY3 and conn_password:
-                # pylint: disable=unexpected-keyword-arg
-                p = subprocess.Popen(cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE, pass_fds=self.sshpass_pipe)
-            else:
-                p = subprocess.Popen(cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            stdin = p.stdin
+            try:
+                if PY3 and conn_password:
+                    # pylint: disable=unexpected-keyword-arg
+                    p = subprocess.Popen(cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE,
+                                         stderr=subprocess.PIPE, pass_fds=self.sshpass_pipe)
+                else:
+                    p = subprocess.Popen(cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE,
+                                         stderr=subprocess.PIPE)
+                stdin = p.stdin
+            except (OSError, IOError):
+                raise AnsibleError(u"Could not find ssh client on the host. Please install it.")
```

Comment:
use the error as this can also happen when sshpass is missing or the client is not executable/unreadable, not just this case.

Reply:
Thank you for the review. Changed exception to the more narrow one: `FileNotFoundError`.